contracts-js/BTCUtils.js

```diff
@@ -0,0 +1,1031 @@
+// /** @title BitcoinSPV */
+// /** @author Summa (https://summa.one) */
+
+// import {BytesLib} from "./BytesLib.sol";
+// import {SafeMath} from "./SafeMath.sol";
+
+// sha256 --> Hold off on this for now
+// ripemd160 --> Hold off on this for now
+// keccack256 --> Can just drop this and do an equality test instead, James will find equality test to use
```

Comment:
Here is a `safeEqual` function that compares buffers in constant time
https://github.com/bcoin-org/bcrypto/blob/91bfabae43135ead6c2b2e43833f3a2325ebd737/lib/safe.js#L46

Reply:
@ErinHales @dominiquejane 

constant-time implementations are amazing in off-chain cryptography. They make it so that you can't learn any secrets by timing how long a computation takes. But unfortunately, they're more expensive than most algorithms, so we probably won't use them for things that run on-chain (where there are no secrets anyway)